### PR TITLE
Move contributing guide details to website and rename two sections

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR moves the detailed contributing sections to the website and renames two sections:
What can I do? -> Ways to Contribute
How can I talk to you? -> Getting Help
A shorter contributing guide with links to the website remains in root dir of the PyGMT repository.

Other suggested revisions will be submitted in a separate PR to keep changes isolated.

Fixes https://github.com/GenericMappingTools/pygmt/issues/1205

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by willschlitzer — APPROVED** [review]
Looks good! I think this was a needed change.

**Review by michaelgrund — APPROVED** [review]

**Review by core-man — APPROVED** [review]
Great~ I cannot see the **Preview** button right now, but it seems that the movement is smooth enough.

File: doc/contributing.md [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+* Make a tutorial or gallery example of how to do something.
+* Provide feedback about how we can improve the project or about your particular use
+  case.
+* Contribute code you already have. It doesn't need to be perfect! We will help you
+  clean things up, test it, etc.
+
+
+## Getting Help
+
+Discussion often happens in the issues and pull requests.
```
Comment (weiji14) [inline doc/contributing.md:27]:
```suggestion
Discussion often happens on GitHub issues and pull requests.
```

**Review by weiji14 — COMMENTED** [review]
Two points to consider:

1. The third and fourth level section headings don't seem to be very clear on the rendered docs. Something about the sphinx theme maybe?

image

compared with the current CONTRIBUTING.md on GitHub which uses bold text.

image

2. There is a "Contributing" link under "Getting Help and Contributing" on the sidebar which points to CONTRIBUTING.md on GitHub. Maybe the "Contributors Guide" (and "Maintainers Guide") should fit under there instead of under "Reference Documentation"?

image

P.S. Not sure why the "Show Deployment" button isn't showing up. But y'all can click on the Vercel deployment details under the checks section to get to the online rendered docs.

image

**Comment by meghanrjones** [comment]
> Two points to consider:
> 
>     1. The third and fourth level section headings don't seem to be very clear on the rendered docs. Something about the sphinx theme maybe?
> 

I think this is a myst parser issue based on comparisons of the changelog when it was .rst versus .md. I assumed that people would prefer markdown over ReST, though it does not really matter to me either way. I would like to submit a separate PR with some more general suggestions for the contributing guide. Would it be OK to see about improving the rendering in that instead?

>     1. There is a "Contributing" link under "Getting Help and Contributing" on the sidebar which points to CONTRIBUTING.md on GitHub. Maybe the "Contributors Guide" (and "Maintainers Guide") should fit under there instead of under "Reference Documentation"? Or we can remove the link to CONTRIBUTING.md 🙂
> 

I'll remove the CONTRIBUTING.md link. Eventually, I would prefer for this to go under a development tab but will leave that change to be reviewed in https://github.com/GenericMappingTools/pygmt/pull/1308.

**Pull request merged**